creusot/tests/should_succeed/iterators/05_map.mlcfg

```diff
@@ -2257,34 +2215,36 @@ module C05Map_Impl0_Next
     goto BB5
   }
   BB4 {
-    assume { Resolve0.resolve _5 };
+    assume { Resolve0.resolve _3 };
+    assert { Inv1.inv self };
     assume { Resolve1.resolve self };
     absurd
   }
   BB5 {
-    v <- Core_Option_Option_Type.some_0 _5;
-    _5 <- (let Core_Option_Option_Type.C_Some a = _5 in Core_Option_Option_Type.C_Some (any Item0.item));
-    assume { Resolve0.resolve _5 };
+    v <- Core_Option_Option_Type.some_0 _3;
+    _3 <- (let Core_Option_Option_Type.C_Some a = _3 in Core_Option_Option_Type.C_Some (any Item0.item));
+    assume { Resolve0.resolve _3 };
     assert { [@expl:assertion] [#"../05_map.rs" 67 16 67 62] Precondition0.precondition (C05Map_Map_Type.map_func ( * self)) (v) };
     goto BB6
   }
   BB6 {
-    _11 <- ([#"../05_map.rs" 68 16 68 55] Ghost.new ());
+    _9 <- ([#"../05_map.rs" 68 16 68 55] Ghost.new ());
     goto BB7
   }
   BB7 {
-    assume { Resolve2.resolve _11 };
-    _14 <- borrow_mut (C05Map_Map_Type.map_func ( * self));
-    self <- { self with current = (let C05Map_Map_Type.C_Map a b =  * self in C05Map_Map_Type.C_Map a ( ^ _14)) };
+    assume { Resolve2.resolve _9 };
+    _12 <- borrow_mut (C05Map_Map_Type.map_func ( * self));
+    self <- { self with current = (let C05Map_Map_Type.C_Map a b =  * self in C05Map_Map_Type.C_Map a ( ^ _12)) };
     goto BB8
   }
   BB8 {
     goto BB9
   }
   BB9 {
+    assert { Inv1.inv self };
```

Comment:
this assertion is tricky to prove

Reply:
Ok, I had a look at this. I think this is not provable in the current state, because the invariant is asserted before the function call. To me, this looks like a bug, because at this program point, the variable is still borrowed, and within the scope (i.e., no STORAGEDEAD), so it should not be resolved at this point.